cmake/Dependencies.cmake

```diff
@@ -1482,7 +1480,17 @@ if(CAFFE2_CMAKE_BUILDING_WITH_MAIN_REPO AND NOT INTERN_DISABLE_ONNX)
     list(APPEND Caffe2_DEPENDENCY_LIBS onnx_proto onnx)
   endif()
   include_directories(${FOXI_INCLUDE_DIRS})
-  list(APPEND Caffe2_DEPENDENCY_LIBS foxi_loader)
+
+  add_definitions(-DONNX_NAMESPACE=${ONNX_NAMESPACE})
+  include_directories(SYSTEM ${PROJECT_SOURCE_DIR}/third_party/optimizer)
+  add_library(onnx_optimizer STATIC
```

Comment:
Why re-defining `onnx_optimizer` library here instead of using one defined in (which can be included using `add_subdirectory("${CAFFE2_THIRD_PARTY_ROOT}/onnx_optimizer" "${CONFU_DEPENDENCIES_BINARY_DIR}/onnx_optimizer")`  https://github.com/onnx/optimizer/blob/a37748b2c3a80dad4274401c45c5026c7a506730/CMakeLists.txt#L23

Reply:
Updating this, now I see the error:
 add_subdirectory given source
  "/var/lib/jenkins/workspace/third_party/onnx_optimizer" which is not an
  existing directory.